packages/runtime/src/adapter/sparql_adapter.ts

```diff
@@ -0,0 +1,84 @@
+import type * as RDF from '@rdfjs/types'
+import { CONSTRUCT, Construct, Describe } from '@tpluscode/sparql-builder'
+import { VALUES } from '@tpluscode/sparql-builder/expressions'
+import * as D from 'docmaps-sdk'
+import * as TE from 'fp-ts/lib/TaskEither'
+import n3 from 'n3'
+import { collect } from 'streaming-iterables'
+import { pipe } from 'fp-ts/lib/pipeable'
+import factory from '@rdfjs/data-model'
+import { Set } from 'immutable'
+import { BackendAdapter } from '../types'
+// import util from 'util'
+
+/** Inteface for an in-memory or over-the-web mechanism that accepts
+ * SPARQL queries and returns triples.
+ * @since 0.1.0
+ */
+export interface SparqlProcessor {
+  // truthy(query: string): Promise<boolean>
+  triples(query: Construct | Describe): TE.TaskEither<Error, AsyncIterable<RDF.Quad>>
+  // bindings(query: string): Promise<AsyncIterable<{ [key: string]: RDF.Term }>>
+}
+
+function FindDocmapQuery(iri: string): Construct | Describe {
+  const subj = factory.namedNode(iri)
+
+  const values = [{ map: subj }]
+
+  // FIXME: use sparql builder more idiomatically
+  const q = CONSTRUCT`
```

Comment:
No need to understand this query but we can discuss SPARQL at a lunch&learn sometime.